okhttp/src/main/kotlin/okhttp3/Cache.kt

```diff
@@ -490,9 +500,14 @@ class Cache internal constructor(
      * optional. If present, it contains the TLS version.
      */
     @Throws(IOException::class) constructor(rawSource: Source) {
-      try {
+      rawSource.use {
         val source = rawSource.buffer()
-        url = source.readUtf8LineStrict()
+        val urlLine = source.readUtf8LineStrict()
+        // Choice here is between failing with a correct RuntimeException
+        // or mostly silently with an IOException
+        url = urlLine.toHttpUrlOrNull() ?: throw IOException("Cache Failure for $urlLine").also {
```

Comment:
3 options here

1) runtime exception - fatal...
2) io exception only
3) io exception and log a warning

Reply:
change `cache failure` to `cache corruption` ?